File: dashboard_v1/src/types/EnrichmentSchema.ts
/**
 * APEX Enrichment Schema v1.0
 * Mirrors backend Python schema exactly
 */

// =============================================================================
// PROFESSIONAL PROFILE
// =============================================================================
export interface CurrentRole {
  title: string;
  company: string;
  tenure?: string;
  responsibilities: string[];
}

export interface CareerTrajectory {
  previous_roles: string[];
  industry_experience: string[];
  expertise_areas: string[];
}

export interface ProfessionalProfile {
  executive_summary: string;
  current_role: CurrentRole;
  career_trajectory: CareerTrajectory;
  education: string[];
  achievements: string[];
  community_involvement: string[];
}

// =============================================================================
// COMPANY INTELLIGENCE
// =============================================================================
export interface CompanyOverview {
  name: string;
  industry: string;
  business_model: string;
  founded?: string;
  headquarters?: string;
  employee_count?: string;
}

export interface CompanyFinancials {
  revenue?: string;
  growth_rate?: string;
  funding?: string;
  key_metrics: string[];
}

export interface MarketPosition {
  target_market: string;
  competitive_advantages: string[];
  competitors: string[];
}

export interface CompanyIntelligence {
  overview: CompanyOverview;
  financials?: CompanyFinancials;
  market_position: MarketPosition;
  recent_news: string[];
  strategic_priorities: string[];
}

// =============================================================================
// SALES INTELLIGENCE
// =============================================================================
export interface PainPoint {
  title: string;
  description: string;
  priority: 'high' | 'medium' | 'low';
}

export interface Opportunity {
  title: string;
  description: string;
  alignment: string;
}

export interface Objection {
  objection: string;
  response: string;
}

export interface SalesIntelligence {
  match_score: number;
  match_reasoning: string;
  pain_points: PainPoint[];
  opportunities: Opportunity[];
  buying_triggers: string[];
  decision_factors: string[];
  objections: Objection[];
  why_now: string;
  why_us: string;
}

// =============================================================================
// PERSONALITY PROFILE
// =============================================================================
export interface MBTIDimension {
  dimension: string;
  preference: string;
  evidence: string;
}

export interface MBTI {
  type: string;
  dimensions: MBTIDimension[];
}

export interface DISC {
  primary: string;
  secondary?: string;
}

export interface CommunicationStyle {
  preferences: string[];
  dos: string[];
  donts: string[];
}

export interface PersonalityProfile {
  mbti?: MBTI;
  disc?: DISC;
  communication_style: CommunicationStyle;
  best_opening_approach: string;
}

// =============================================================================
// OUTREACH ASSETS
// =============================================================================
export interface CallScript {
  level: number;
  script: string;
}

export interface EmailTemplate {
  type: 'initial' | 'followup' | 'breakup';
  subject: string;
  body: string;
}

export interface OutreachAssets {
  talking_points: string[];
  call_scripts: CallScript[];
  email_templates: EmailTemplate[];
  linkedin_message?: string;
  voicemail_script?: string;
}

// =============================================================================
// MAIN ENRICHMENT DATA
// =============================================================================
export interface EnrichmentData {
  version: string;
  generated_at: string;
  professional: ProfessionalProfile;
  company: CompanyIntelligence;
  sales: SalesIntelligence;
  personality: PersonalityProfile;
  outreach: OutreachAssets;
}

// =============================================================================
// CONTACT
// =============================================================================
export interface Contact {
  id: string;
  hubspot_id?: string;
  first_name: string;
  last_name: string;
  email?: string;
  phone?: string;
  title?: string;
  company?: string;
  enrichment?: EnrichmentData;
  created_at: string;
  updated_at: string;
  enriched_at?: string;
}

File: .backup-20251214-154343/apps/backend/schemas/enrichment_schema.py
"""
APEX Enrichment Schema v1.0
Single source of truth for all enrichment data structures
"""
from typing import Optional, List, Literal
from pydantic import BaseModel
from datetime import datetime

# =============================================================================
# PROFESSIONAL PROFILE
# =============================================================================
class CurrentRole(BaseModel):
    title: str
    company: str
    tenure: Optional[str] = None
    responsibilities: List[str] = []

class CareerTrajectory(BaseModel):
    previous_roles: List[str] = []
    industry_experience: List[str] = []
    expertise_areas: List[str] = []

class ProfessionalProfile(BaseModel):
    executive_summary: str
    current_role: CurrentRole
    career_trajectory: CareerTrajectory
    education: List[str] = []
    achievements: List[str] = []
    community_involvement: List[str] = []

# =============================================================================
# COMPANY INTELLIGENCE
# =============================================================================
class CompanyOverview(BaseModel):
    name: str
    industry: str
    business_model: str
    founded: Optional[str] = None
    headquarters: Optional[str] = None
    employee_count: Optional[str] = None

class CompanyFinancials(BaseModel):
    revenue: Optional[str] = None
    growth_rate: Optional[str] = None
    funding: Optional[str] = None
    key_metrics: List[str] = []

class MarketPosition(BaseModel):
    target_market: str
    competitive_advantages: List[str] = []
    competitors: List[str] = []

class CompanyIntelligence(BaseModel):
    overview: CompanyOverview
    financials: Optional[CompanyFinancials] = None
    market_position: MarketPosition
    recent_news: List[str] = []
    strategic_priorities: List[str] = []

# =============================================================================
# SALES INTELLIGENCE
# =============================================================================
class PainPoint(BaseModel):
    title: str
    description: str
    priority: Literal['high', 'medium', 'low'] = 'medium'

class Opportunity(BaseModel):
    title: str
    description: str
    alignment: str  # How your product helps

class Objection(BaseModel):
    objection: str
    response: str

class SalesIntelligence(BaseModel):
    match_score: int  # 0-100
    match_reasoning: str
    pain_points: List[PainPoint] = []
    opportunities: List[Opportunity] = []
    buying_triggers: List[str] = []
    decision_factors: List[str] = []
    objections: List[Objection] = []
    why_now: str
    why_us: str

# =============================================================================
# PERSONALITY PROFILE
# =============================================================================
class MBTIDimension(BaseModel):
    dimension: str
    preference: str
    evidence: str

class MBTI(BaseModel):
    type: str  # e.g., "ENTJ"
    dimensions: List[MBTIDimension] = []

class DISC(BaseModel):
    primary: str  # e.g., "D - Dominant"
    secondary: Optional[str] = None

class CommunicationStyle(BaseModel):
    preferences: List[str] = []
    dos: List[str] = []
    donts: List[str] = []

class PersonalityProfile(BaseModel):
    mbti: Optional[MBTI] = None
    disc: Optional[DISC] = None
    communication_style: CommunicationStyle
    best_opening_approach: str

# =============================================================================
# OUTREACH ASSETS
# =============================================================================
class CallScript(BaseModel):
    level: int  # 1, 2, 3
    script: str

class EmailTemplate(BaseModel):
    type: Literal['initial', 'followup', 'breakup']
    subject: str
    body: str

class OutreachAssets(BaseModel):
    talking_points: List[str] = []
    call_scripts: List[CallScript] = []
    email_templates: List[EmailTemplate] = []
    linkedin_message: Optional[str] = None
    voicemail_script: Optional[str] = None

# =============================================================================
# MAIN ENRICHMENT DATA STRUCTURE
# =============================================================================
class EnrichmentData(BaseModel):
    version: str = "1.0"
    generated_at: str  # ISO timestamp
    professional: ProfessionalProfile
    company: CompanyIntelligence
    sales: SalesIntelligence
    personality: PersonalityProfile
    outreach: OutreachAssets

# =============================================================================
# CONTACT MODEL
# =============================================================================
class Contact(BaseModel):
    id: str
    hubspot_id: Optional[str] = None
    first_name: str
    last_name: str
    email: Optional[str] = None
    phone: Optional[str] = None
    title: Optional[str] = None
    company: Optional[str] = None
    enrichment: Optional[EnrichmentData] = None
    created_at: datetime
    updated_at: datetime
    enriched_at: Optional[datetime] = None
